app/scripts/lib/ppom/ppom-util.ts

```diff
@@ -183,6 +186,17 @@ function normalizePPOMRequest(
   };
 }
 
+function sanitizeRequest(request: JsonRpcRequest) {
+  if (
+    request.method === METHOD_SIGN_TYPED_DATA_V4 ||
+    request.method === METHOD_SIGN_TYPED_DATA_V3
+  ) {
+    if (Array.isArray(request.params)) {
+      request.params = request.params.slice(0, 2);
+    }
+  }
+}
```

Comment:
possibly a comment to mention why we are explicitly reading only the first two params might be a good idea here for context

Reply:
Done